src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -98,8 +98,17 @@ fun resolvePath(path: RsPath, lookup: ImplLookup = ImplLookup.relativeTo(path)):
 
         val assocTypes = run {
             if (element is RsTraitItem) {
+                // Iterator<Item=T>
+                val assocBindings =  path.typeArgumentList?.assocTypeBindingList?.mapNotNull {
+                    val name = it.name
+                    val type = it.typeReference.type
+                    element.associatedTypesTransitively.find { it.name == name }?.let {
```

Comment:
Is it ok that we check name equality for search instead of using `resolve` or something else?